table/tables/tables.go

```diff
@@ -113,9 +121,46 @@ func TableFromMeta(alloc autoid.Allocator, tblInfo *model.TableInfo) (table.Tabl
 	}
 
 	t.meta = tblInfo
+	partitionExpr, err := generatePartitionExpr(tblInfo)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	t.partitionExpr = partitionExpr
 	return t, nil
 }
 
+func generatePartitionExpr(tblInfo *model.TableInfo) ([]expression.Expression, error) {
+	pi := tblInfo.GetPartitionInfo()
+	if pi == nil {
+		return nil, nil
+	}
+	// TODO: Support other partition method.
+	if pi.Type != model.PartitionTypeRange {
+		return nil, nil
```

Comment:
unsupport type maybe better return unsupport error now to protect user miss user before our support~?